pkg/operator/controllers/autosizednodes/autosizednodes_controller.go

```diff
@@ -90,7 +94,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, fmt.Errorf("could not fetch KubeletConfig: %w", err)
 	}
 
-	// If already exists, update the spec
+	// If already exists, update the spec and annotations
+	if config.Annotations == nil {
+		config.Annotations = defaultConfig.Annotations
+	}
+	if val, ok := config.Annotations[mcAnnotationName]; !ok || val != mcAnnotationValue {
+		config.Annotations[mcAnnotationName] = mcAnnotationValue
```

Comment:
What happens if a different `kubeletconfig` (maybe one provided by the customer) already his this annotation and value?